scrapy/middleware.py

```diff
@@ -1,9 +1,13 @@
 import logging
 import pprint
 from collections import defaultdict, deque
-from typing import Callable
+from typing import Callable, Dict, Deque
 
+from twisted.internet import defer
```

Comment:
What about `from twisted.internet.defer import Deferred` as you did in the first file? I would rather use `Deferred` than `defer.Deferred`. I would even add such an import to `scrapy.utils.defer`, which was already importing `defer`.

Reply:
`scrapy.utils.defer` is harder as it uses `succeed` and `fail` and I don't like aliases or importing these names as is.

Reply:
Makes sense to me, but still it may be best to have both imports than to use `defer.Deferred` everywhere.

Reply:
Ah yeah, I didn't think about using both.